go/workflow/couler/codegen.go

```diff
@@ -199,16 +210,15 @@ func GenCode(programIR []ir.SQLFlowStmt, session *pb.Session) (string, error) {
 
 // GenYAML translate the Couler program into Argo YAML
 func GenYAML(coulerProgram string) (string, error) {
-	cmdline := bytes.Buffer{}
-	fmt.Fprintf(&cmdline, "couler run --mode argo --workflow_name sqlflow ")
-	if c := os.Getenv("SQLFLOW_WORKFLOW_CLUSTER_CONFIG"); len(c) > 0 {
-		fmt.Fprintf(&cmdline, "--cluster_config %s ", c)
+	file, e := ioutil.TempFile("/tmp", "sqlflow.py")
+	if e != nil {
+		return "", e
 	}
-	fmt.Fprintf(&cmdline, "--file -")
-
-	coulerExec := strings.Split(cmdline.String(), " ")
-	// execute command: `cat sqlflow.couler | couler run --mode argo --workflow_name sqlflow --file -`
-	cmd := exec.Command(coulerExec[0], coulerExec[1:]...)
+	defer os.Remove(file.Name())
+	if e := ioutil.WriteFile(file.Name(), []byte(coulerProgram), 0644); e != nil {
+		return "", e
+	}
+	cmd := exec.Command("python", file.Name())
```

Comment:
Now that https://github.com/couler-proj/couler/pull/106 has been merged. You can try switching back to `python -u` here.

Reply:
After testing the last code base, `python -u` would generate an Argo YAML with `generateName`:
``` yaml
    generateName: stdin-
```

the `stdin` prefix is too general that we can not distinguish with other workflows.

Reply:
Update: we discussed this offline and we have a workaround for this for now. We can consider allowing users to configure the prefix of `generateName` in `couler.config_workflow()` later on.